utils/call-gen/src/arbitrary_call.rs

```diff
@@ -0,0 +1,197 @@
+const MAX_PAYLOAD_SIZE: usize = 512 * 1024;
+static_assertions::const_assert!(MAX_PAYLOAD_SIZE <= gear_core::message::MAX_PAYLOAD_SIZE);
+
+#[derive(Debug, Clone)]
+/// New-type wrapper over array of [`GearCall`]s.
+///
+/// It's main purpose is to be an implementor of `Arbitrary` for the array of [`GearCall`]s.
+/// New-type is required as array is always a foreign type.
+pub struct GearCalls(pub [GearCall; GearCalls::MAX_CALLS]);
+
+impl GearCalls {
+    pub const MAX_CALLS: usize = GearCalls::INIT_MSGS + GearCalls::HANDLE_MSGS;
+    pub const INIT_MSGS: usize = 10;
+    pub const HANDLE_MSGS: usize = 25;
+}
+
+impl<'a> Arbitrary<'a> for GearCalls {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+        // Newline to easily browse logs.
+        println!("");
+
+        log::trace!("New GearCalls generation: random data received {}", u.len());
+
+        // 25 MiB is an approximate assessment for 35 calls,
+        // where each call of:
+        // 1. `UploadProgram` requires 1024 KiB for payload an salt and 50 KiB for code.
+        // 2. `SendMessage` requires 512 Kib for payload.
+        // So, [10 * (1024 + 50)] + [25 * 512] = 23540 KiB
+        if u.len() < 25_000_000_usize {
+            log::trace!("Not enough bytes for creating gear calls");
+            return Err(arbitrary::Error::NotEnoughData);
+        }
+
+        let log_data = format!(
+            "Generated from corpus - {}",
+            get_sha1_string(u.peek_bytes(u.len()).expect("checked"))
+        );
+        let gas = 249_000_000_000_u64;
+        let value = 0;
+        let mut programs = [ProgramId::default(); GearCalls::INIT_MSGS];
+        // Upload code used as a default value.
+        let mut calls = get_uninitialized_calls();
+
+        // Generate `GearCalls::INIT_MSGS` number of `UploadProgram` calls.
+        for i in 0..GearCalls::INIT_MSGS {
+            log::trace!("New gear-wasm generation");
+
+            log::trace!("Random data before wasm gen {}, iter - {i}", u.len());
+
+            let code = gear_wasm_gen::generate_gear_program_code(
+                u,
+                config(programs, Some(log_data.clone())),
+            )?;
+            log::trace!("Random data after wasm gen {}, iter - {i}", u.len());
+            log::trace!("Code length {:?}", code.len());
+
+            let salt = arbitrary_payload(u)?;
+            log::trace!("Random data after salt gen {}, iter - {i}", u.len());
+            log::trace!("Salt length {:?}", salt.len());
+
+            let payload = arbitrary_payload(u)?;
+            log::trace!(
+                "Random data after payload (upload_program) gen {}, iter - {i}",
+                u.len()
+            );
+            log::trace!("Payload (upload_program) length {:?}", payload.len());
+
+            programs[i] = ProgramId::generate(CodeId::generate(&code), &salt);
+            calls[i] = MaybeUninit::new(GearCall::UploadProgram(UploadProgramArgs((
```

Comment:
`calls[i].write(GearCall::UploadProgram(UploadProgramArgs(...))` here and on other places

Reply:
done